File: insertion_sort.py
import random

def insertion_sort(arr):
	for i in range(1, len(arr)):
		temp = arr[i]

		while i > 0 and arr[i - 1] > temp:
			arr[i], arr[i - 1] = arr[i - 1], arr[i]
			i -= 1

	return arr

if __name__ == "__main__":
	arr = []
	
	for i in range(10):
		arr.append(random.randint(1, 20))

	print(arr)
	print(insertion_sort(arr))

File: js/insertion_sort.js
function insertionSort(arr) {
	for (let i = 1; i < arr.length; i++) {
		let temp = arr[i];
		let pos = i - 1;

		while ((pos > -1) && (temp < arr[pos])) {
			arr[pos + 1] = arr[pos];
			pos--;
		}

		arr[pos + 1] = temp;
	}

	return arr;
}

let unsorted = [5, 2, 4, 6, 1, 3];
console.log(unsorted);
console.log(insertionSort(unsorted));